### PR TITLE
planer: tiny refine code to return as early as possible In convertToIndexScan and convertToTableScan.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Not a problem, just to make the code to return as early as possible 

### Check List 

Tests 

 - No code: no code logic change

Code changes

Side effects

Related changes

### CODE REVIEW

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -619,6 +621,9 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 	if prop.TaskTp == property.CopDoubleReadTaskType {
 		return invalidTask, nil
 	}
+	if !prop.IsEmpty() && !candidate.isMatchProp {
```
Comment (crazycs520) [inline planner/core/find_best_task.go]: One more question: 
* The check use for what ?
* Why `convertToIndexScan` doesn't have this check?

**Comment by zz-jason** [comment]
@crazycs520 please fix ci

File: planner/core/find_best_task.go [hunk 2]
```diff
@@ -519,8 +524,6 @@ func (ds *DataSource) convertToIndexScan(prop *property.PhysicalProperty, candid
 		expectedCnt := math.MaxFloat64
```
Comment (eurekaka) [inline planner/core/find_best_task.go]: Remove this line and the if branch as well?

Comment (crazycs520) [reply planner/core/find_best_task.go]: em... how to remove this and keep the logic remain same?

Comment (eurekaka) [reply planner/core/find_best_task.go]: The `expectedCnt` should always be set to `prop.ExpectedCnt` now, i.e, the `prop.IsEmpty()` would always return true when coming to here, so we can just use `prop.ExpectedCnt` as parameter of `addPushedDownSelection`?

Comment (crazycs520) [reply planner/core/find_best_task.go]: I got you, Thanks~

File: planner/core/find_best_task.go [hunk 3]
```diff
@@ -516,13 +521,9 @@ func (ds *DataSource) convertToIndexScan(prop *property.PhysicalProperty, candid
...
-		expectedCnt := math.MaxFloat64
-		if prop.IsEmpty() {
-			expectedCnt = prop.ExpectedCnt
-		} else {
-			return invalidTask, nil
-		}
-		is.addPushedDownSelection(cop, ds, expectedCnt, path)
+		// prop.IsEmpty() would always return true when coming to here,
+		// so we can just use prop.ExpectedCnt as parameter of addPushedDownSelection.
+		is.addPushedDownSelection(cop, ds, prop.ExpectedCnt, path)
```
Comment (eurekaka) [inline planner/core/find_best_task.go]: We can further more extract `addPushedDownSelection` out of the if-else branch now?

Comment (crazycs520) [reply planner/core/find_best_task.go]: done.

**Pull request merged**